### PR TITLE
The Great Species Dedatumming 2: Makes external organs operable

### PR DESCRIPTION
**Pull request by Time-Green**
## About The Pull Request

Finally makes external organs (horns, moth wings, antennae etc.) operable!

Here's a deconstructed lizard:
image

Here's an enhanced human:
image

Here is the future that liberals want:
image

Also adds a new surgery: feature manipulation. It's a shorter version of organ manipulation that only lets you operate on species features. Note that organ manipulation can operate only on internal organs now. 

## Why It's Good For The Game

Immovable mechanics are not fun. What's the point of having all these cool cosmetic features if they can't be surgically removed and given to someone else?

Encourages creativity and mad science. Moves us closer to complete species dedatumization

## Changelog

:cl: Time-Green
add: You can now remove/add all those ugly species features with the power of surgery! A lizards frills, horns, snout and spines, a moths wings and antennae, podperson hair and just wings in general
add: Species features can be altered by organ manipulation or the new 'feature manipulation' surgery! 'Feature manipulation' is a shorter version of organ manip that let's you only operate on external features
balance: Organ manipulation only works on internal organs now. Use 'feature manipulation' for external organs.
code: cleans up magic numbers in surgery
/:cl:

Species change permission-slip:
image

### CODE REVIEW

File: code/modules/surgery/organ_manipulation.dm [hunk 1]
```diff
@@ -176,3 +192,13 @@
 				span_notice("[user] can't seem to extract anything from [target]'s [parse_zone(target_zone)]!"),
 				span_notice("[user] can't seem to extract anything from [target]'s [parse_zone(target_zone)]!"))
 	return FALSE
+
+/datum/surgery_step/manipulate_organs/proc/can_use_organ(mob/user, obj/item/organ/organ)
+	return TRUE
+
+/datum/surgery_step/manipulate_organs/external
+	time = 32
```
Comment (Fikou) [inline code/modules/surgery/organ_manipulation.dm:200]: use SECONDS defines

File: code/modules/surgery/organ_manipulation.dm [hunk 2]
```diff
@@ -102,6 +111,10 @@
 		if(!meatslab.useable)
 			to_chat(user, span_warning("[target_organ] seems to have been chewed on, you can't use this!"))
 			return -1
+
+		if(!can_use_organ(user, meatslab))
+			return -1
```
Comment (Fikou) [inline code/modules/surgery/organ_manipulation.dm:116]: can you make this a define instead of a magic number?

**Comment by CursedBirb** [comment]
As for now only ears, tails and wings were possible to obtain with wings having no sprite(they also could not get removed so you needed to butcher moth to get invisible wings that gains sprite once used on someone with ogran manipulation)

**Comment by Sightld2** [comment]
As an Ai main, how do I know who is human and who is expendable?

**Comment by Time-Green** [comment]
> As an Ai main, how do I know who is human and who is expendable?

I think it’ll still be pretty obvious who is non-human, since this doesn’t change their skin. And in the end you can always health-analyze them

**Comment by dragomagol** [comment]
Wildly cursed 👍

File: code/modules/surgery/organs/external/_external_organs.dm [hunk 3]
```diff
@@ -8,7 +8,7 @@
 	desc = "An external organ that is too external."
 
 	///Unremovable is until the features are completely finished
```
Comment (GoldenAlpharex) [inline code/modules/surgery/organs/external/_external_organs.dm:10]: You forgot to change this comment, since they're no longer going to be unremovable.

Comment (Time-Green) [reply code/modules/surgery/organs/external/_external_organs.dm]: Woops, updated

**Comment by MrDoomBringer** [comment]
Huge

**Comment by Farquaar** [comment]
If sci-fi and superhero cartoons have taught me anything, it's that we'll soon see epic battles between the "mutes" and the "norms" envelop the station

**Comment by Dorsisdwarf** [comment]
Being able to remove a lizards whole face leaving a human face underneath seems off.

**Comment by Time-Green** [comment]
> Being able to remove a lizards whole face leaving a human face underneath seems off.

My headcannon is that it’s because snouts are not actually organs and the doctors just leave it likr that.

But yeah that’s just how it’s coded and sprited, I think it’s funny enough but tell me if you got a better solution

File: code/modules/surgery/organs/external/_external_organs.dm [hunk 4]
```diff
@@ -144,6 +144,11 @@
 	var/mutable_appearance/appearance = mutable_appearance(sprite_datum.icon, finished_icon_state, layer = -image_layer)
 	appearance.dir = image_dir
 
+	///Also give the icon to the obj
+	if(use_mob_sprite_as_obj_sprite)
+		icon_state = finished_icon_state
+		icon = sprite_datum.icon
```
Comment (Fikou) [inline code/modules/surgery/organs/external/_external_organs.dm:150]: wont this give it directional sprites? stuff like moth wings should always be facing north probably

Comment (Time-Green) [reply code/modules/surgery/organs/external/_external_organs.dm:150]: Oops, fixed

File: code/modules/surgery/organs/external/_external_organs.dm [hunk 5]
```diff
@@ -230,9 +235,16 @@
 /obj/item/organ/external/proc/override_color(rgb_value)
 	CRASH("External organ color set to override with no override proc.")
 
+///use_mob_sprite_as_obj_sprite uses four directional mob sprites, which is bad for objects so just dont let them change dir and we good
+/obj/item/organ/external/setDir(newdir)
+	return ..(use_mob_sprite_as_obj_sprite ? SOUTH : newdir)
```
Comment (Watermelon914) [inline code/modules/surgery/organs/external/_external_organs.dm:240]: Edit the icon to only have a south state instead, this is really hacky

File: code/modules/surgery/organ_manipulation.dm [hunk 6]
```diff
@@ -176,3 +192,13 @@
 				span_notice("[user] can't seem to extract anything from [target]'s [parse_zone(target_zone)]!"),
 				span_notice("[user] can't seem to extract anything from [target]'s [parse_zone(target_zone)]!"))
 	return FALSE
+
+/datum/surgery_step/manipulate_organs/proc/can_use_organ(mob/user, obj/item/organ/organ)
+	return TRUE
```
[the remark this answers is not in the document]
Comment (Watermelon914) [reply code/modules/surgery/organ_manipulation.dm:197]: Make a new subtype that the surgery before uses, where it only affects internal organs

File: code/__DEFINES/surgery.dm [hunk 7]
```diff
@@ -0,0 +1,25 @@
+/// Flags for the organ_flags var on /obj/item/organ. Binary 0 is skipped to avoid accidental features when someone forgets to set organ flags
+#define ORGAN_SYNTHETIC (1<<1) //Synthetic organs, or cybernetic organs. Reacts to EMPs and don't deteriorate or heal
+#define ORGAN_FROZEN (1<<1) //Frozen organs, don't deteriorate
+#define ORGAN_FAILING (1<<3) //Failing organs perform damaging effects until replaced or fixed
+#define ORGAN_INTERNAL (1<<4) //majority of organs, like hearts, lungs, livers, etc
+#define ORGAN_EXTERNAL (1<<5) //Was this organ implanted/inserted/etc, if true will not be removed during species change.
```
Comment (Watermelon914) [inline code/__DEFINES/surgery.dm:6]:
```suggestion
/// An organ that is considered external (e.g. wings)
#define ORGAN_EXTERNAL (1<<5)
```

**Comment by Watermelon914** [comment]
edited your description to be more correct

**Comment by Time-Green** [comment]
> edited your description to be more correct

Thanks! I forgot to update it

**Pull request merged**